util/util.go

```diff
@@ -197,6 +197,16 @@ func GetControlPlaneMachinesFromList(machineList *clusterv1.MachineList) (res []
 	return
 }
 
+// IsExternalManagedControlPlane returns a bool indicating whether the control plane referenced
+// in the passed Unstructured resource is an externally managed control plane such as AKS, EKS, GKE, etc.
+func IsExternalManagedControlPlane(controlPlane *unstructured.Unstructured) bool {
+	managed, found, err := unstructured.NestedBool(controlPlane.Object, "status", "externalManagedControlPlane")
```

Comment:
Should we document somewhere that manage control plane implementation should have this field (might be in  https://cluster-api.sigs.k8s.io/developer/architecture/controllers/control-plane.html#required-status-fields)

Reply:
+1 this isn't part of the contract currently AFAIK

cc @alexeldeib

Reply:
+1 updating the documentation, just a note though the field is completely optional and there is a fallback to the current behavior during delete

Reply:
Makes sense. What are thoughts on making this new field be an optional part of the contract to avoid a breaking change?

Reply:
This is an unusual case in that the field is really required for managed control planes like AKS, EKS, GKE, etc otherwise the Machine controller will suffer this pretty bad bug (instance termination without draining nodes), but not required at all for KubeadmControlPlane. No other optional status fields currently appear to modify functionality and are purely informational. If we make it a new required status field though, I think this would need to wait for a new minor version to be released. I'd like to get the bugfix released sooner rather than later if possible.

Reply:
I'm 👍 to optional. Please update the control plane architecture doc linked above and add it to the list of optional status fields, assuming everyone else agrees.

Reply:
Added a blurb to the docs:
https://github.com/kubernetes-sigs/cluster-api/pull/3673/files#diff-cd671ed1b58bf2cf9d0a69033c9a764dR181-R183